src/components/TicTacToe.jsx

```diff
@@ -48,135 +49,104 @@ const style = {
   }
 };
 
+const Box = ({ style, onClick, children }) => {
+  return (
+    <li
+      style={style}
+      onClick={onClick}
+    >{children}</li>
+  )
+}
+
 class TicTacToe extends Component {
   constructor(props) {
     super(props);
-    // CONSIDER: turn the box default state to an object consisting of { value: '', lock: false }
-    // to prevent mutation of already set boxes
+
     this.state = {
-      boxOneValue: '',
-      boxTwoValue: '',
-      boxThreeValue: '',
-      boxFourValue: '',
-      boxFiveValue: '',
-      boxSixValue: '',
-      boxSevenValue: '',
-      boxEightValue: '',
-      boxNineValue: '',
-      win: false,
-      playerSwitch: true
+      box: ['','','','','','','','',''],
+      winner: '',
+      numberOfPlay: 0,
     }
+    this.positionOne = [];
+    this.positionTwo = [];
     this.handleClick = this.handleClick.bind(this);
     this.handleReset = this.handleReset.bind(this);
-    this.setBoxValue = this.setBoxValue.bind(this);
+    this.handleUpdateBoxValues = this.handleUpdateBoxValues.bind(this);
   }
 
   /**
    * Sets the value of the player game data into the selected box 
    * position.
-   * @param {string} boxPosition the selected box position where game data should be inserted
+   * @param {string} currentPosition the selected box position where game data should be inserted
    * @returns {string} 
    * @memberof TicTacToe
    */
-  setBoxValue(boxPosition) {
-    const { playerSwitch } = this.state;
-    const boxPositionInput = `box${boxPosition}Input`;
-    this[boxPositionInput].innerHTML = playerSwitch ? 'X' : 'O';
+  handleUpdateBoxValues(currentPosition) {
+    let boxCopy, boxFilledValue;
+    const { box, playerSwitch, numberOfPlay, winner } = this.state;
+    // let boxFilledValue = numberOfPlay % 2 ? 'X' : 'O';
+    if (numberOfPlay % 2 === 0) {
+      boxFilledValue = 'X';
+      this.positionOne.push(currentPosition);
```

Comment:
First why are you using na instance variable.
Secondly why push into an already existing value. react encourages immutability and you should embody this in all the apps you are building